internal/controller/certificates/policies/checks.go

```diff
@@ -580,23 +578,20 @@ func SecretBaseLabelsMismatch(input Input) (string, string, bool) {
 	return SecretManagedMetadataMismatch, fmt.Sprintf("wrong base label %s value %q, expected \"true\"", cmapi.PartOfCertManagerControllerLabelKey, value), true
 }
 
-// SecretCertificateDetailsAnnotationsMismatch - When the certificate details annotations are
-// not matching, the secret is updated.
-// NOTE: The presence of the certificate details annotations is checked
-// by the SecretManagedLabelsAndAnnotationsManagedFieldsMismatch function.
+// SecretCertificateDetailsAnnotationsMismatch returns a validation violation when
+// annotations on the Secret do not match the details of the x509 certificate that
+// is stored in the Secret. This function will only compare the annotations that
+// already exist on the Secret and are also present in the certificate metadata.
+// NOTE: Missing and extra annotations are detected by the SecretManagedLabelsAndAnnotationsManagedFieldsMismatch
+// function instead.
 func SecretCertificateDetailsAnnotationsMismatch(input Input) (string, string, bool) {
 	dataAnnotations, err := certificateDataAnnotationsForSecret(input.Secret)
 	if err != nil {
 		return InvalidCertificate, fmt.Sprintf("Failed getting secret annotations: %v", err), true
 	}
 
-	for k, v := range dataAnnotations {
-		existing, ok := input.Secret.Annotations[k]
-		if !ok || existing == v {
-			continue
-		}
-
-		return SecretManagedMetadataMismatch, fmt.Sprintf("Secret metadata %s does not match certificate metadata %s", input.Secret.Annotations[k], v), true
+	if match, key := mapsHaveMatchingValues(dataAnnotations, input.Secret.Annotations); !match {
+		return SecretTemplateMismatch, fmt.Sprintf("Secret metadata %s does not match certificate metadata %s", input.Secret.Annotations[key], dataAnnotations[key]), true
```

Comment:
The first encountered mis-matched key was inconsistent here and depended on the ordering of the map's keys.